src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
     });
 }
 
+fn inlay_hints_from_editor_code_lens(
+    meta: EditorMeta,
+    mut lenses: Vec<CodeLens>,
+    ctx: &mut Context,
+) {
+    let document = match ctx.documents.get(&meta.buffile) {
+        Some(document) => document,
+        None => {
+            return;
+        }
+    };
+    lenses.sort_by_key(|lens| lens.range.start);
+
+    let inlay_hints_from_lenses = lenses
+        .iter()
+        .map(|lens| {
+            let label = lens.command.as_ref().map_or("", |v| &v.title);
+            if label.trim().len() > 0 {
+                let position =
+                    lsp_position_to_kakoune(&lens.range.start, &document.text, ctx.offset_encoding);
+                let line = position.line;
+                let col = position.column;
+                editor_quote(&format!(
+                    "{line}.{col}+0|{{InlayHint}}[\u{1F50E} {}] {{InlayHint}}",
+                    escape_tuple_element(label)
+                ))
+            } else {
+                String::new()
+            }
+        })
+        .join(" ");
+    let command = format!(
+        "set-option buffer lsp_inlay_hints {} {}",
+        meta.version, inlay_hints_from_lenses
```

Comment:
(optional) what do you think about disabling all virtual text (from inlay hints or code lenses) while in insert mode, like we already do for inlay diagnostics?

Reply:
Virtual text should be _on_ during insert mode I think. 

While I'm doing the actual code editing I actually want to know the types even more...